### PR TITLE
Feature/update procs material in ruby blocks

### PR DESCRIPTION
**Pull request by binarygit**
- [ yes] You have read [The Odin Projects Contributing Guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md).

#### 1.Describe the changes made:
Three main additions made to Blocks unit
1. Add a SO article to clarify block capture
2. Show a new method of creating a proc
3. Add another helpful argument property of procs

#### 2. If this PR is related to an open issue please reference it with the `#` operator and the issue number below:

#XXXXX

### CODE REVIEW

**Comment by twalton83** [comment]
Are you still interested in making the requested changes?

**Comment by binarygit** [comment]
Hey, sorry guys I have been away for sometime. I updated the document, I hope it addresses the issues. Let me know please. I won't take so long next time!

File: ruby_programming/advanced_ruby/blocks.md [hunk 1]
```diff
@@ -310,6 +319,20 @@ a_proc.call("apple")
...
+which is also why this is possible:
+
+~~~ruby
+nested_array = [[1, 2], [3, 4], [5, 6]]
+nested_array.select {|a, b| a + b > 10 }
+
+# => [5, 6]
+~~~
+
+As you can see, `#select` has two arguments specified `|a, b|`, on each iteration we pass a single element of nested_array into the block. On the first iteration this is: `[1, 2]`, this array now, is deconstructed automatically (into a = 1, b = 2) and its values compared as specified. So on to the next rounds of iteration in which we pass `[3,4]`, `[7, 8]`, and `[5, 6]` one by one.
```
Comment (CouchofTomato) [inline ruby_programming/advanced_ruby/blocks.md:331]: Sorry to be pedantic but can be make sure the `[3,4]` array has the same spacing as the other ones

**Review by CouchofTomato — CHANGES_REQUESTED** [review]

**Comment by CouchofTomato** [comment]
Hey @binarygit 

Thanks for your work on this. There is just one small change please and then we can get this merged.

**Review by CouchofTomato — APPROVED** [review]

**Comment by CouchofTomato** [comment]
Thanks for the contribution @binarygit

**Pull request merged**